### DISCUSSION TITLE
Cart Issue - products not augmented

### DISCUSSION THREAD

**Discussion by mathamel** [duncanmcclean/simple-commerce#1145] [Help] [answered]
We have a multisite website, using database for all Entries / Collections.

We added the Simple Commerce to it, apply the database  command / migration as per documentation.

Now in cart, when listing item, we have only the product id, cannot display the product name, image or other information.
The checkout form for shipping also doesn't load the 'states / province' related drop down when changing country.

We cannot complete checkout for now.

**Comment by duncanmcclean** [comment]
What does your template look like?

**Reply by mathamel** [reply]
I have this for now:
```


<section class="fluid-container pt-12 text-center">
    {{ partial:typography/prose as="article" class="contents" }}
        <h1>Votre panier</h1>
    {{ /partial:typography/prose }}
</section>

<main>
    {{# collection:products }}

    <div>{{title}} </div>
 
{{ /collection:products #}}
    <div class="max-w-4xl mx-auto my-12 px-6">
        {{ if {sc:cart:has} === false || {sc:cart:count} === 0 }}
            <p class="text-center">Votre panier est vide! <a href="/boutique">Commencer à magaziner.</a>.</p>
        {{ else }}
       
            {{ sc:cart }}
                <table class="bg-white table-auto w-full rounded-lg p-4 mb-4">
                    <thead>
                        <tr class="border-b border-gray-200">
                            <th class="px-6 py-3 text-left font-semibold text-sm w-2/3">Product</th>
                            <th class="px-6 py-3 text-left font-semibold text-sm">Price</th>
                            <th class="px-6 py-3 text-left font-semibold text-sm">Quantity</th>
                            <th class="px-6 py-3 text-right font-semibold text-sm">Total</th>
                        </tr>
                    </thead>

                    <tbody>
                        
                        {{ items }}
                            {{ collection:products id:is="{product}"}}

                          
                                <tr class="border-b border-gray-200">
                                    <td class="text-sm px-6 py-4">
                                        <div class="flex flex-row items-center">
                                            <a class="hidden md:block" href="{{ url }}">
                                                {{ photos }}
                                                    {{ if first }}
                                                        <img class="w-32"
                                                            src="{{ url }}" alt="{{ alt }}">
                                                    {{ /if }}
                                                {{ /photos }}
                                            </a>
                                            <div class="md:px-6">
                                                <a class="font-bold" href="{{ url }}">
                                                    {{ title }}
                                                    {{ if variant }}({{ variant:variant }}){{ /if }}
                                                </a>
    
                                                <br>
    
                                                {{ sc:cart:removeItem :item="id" }}
                                                    <button class="mt-1 block text-sm text-gray-700">Remove</button>
                                                {{ /sc:cart:removeItem }}
                                            </div>
                                        </div>
                                    </td>
                                    <td class="text-sm px-6 py-4">{{ variant:price ?? price }}</td>
                                    <td class="text-sm px-6 py-4">
                                        {{ sc:cart:updateItem :item="id" }}
                                            <select
                                                class="bg-white p-1 rounded w-16"
                                                name="quantity"
                                                onchange="this.parentElement.submit()"
                                            >
                                                {{ loop from="1" to="5" }}
                                                    <option
                                                        value="{{ value }}"
                                                        {{ if quantity == value }} selected {{ /if }}
                                                    >
                                                        {{ value }}
                                                    </option>
                                                {{ /loop }}
                                            </select>
                                        {{ /sc:cart:updateItem }}
                                    </td>
    
                                    <td class="text-sm px-6 py-4 text-right">{{ total }}</td>
                                </tr>    
                            
                            {{ /collection:products }}
                            {{# collection:products as="myproduct"}}
                                {{ if no_results }}
                                    Aucun article trouvé
                                {{ else }}
                                <tr><td colspan="4">{{myproduct:title}} here {{product}}</td></tr>
                                {{ /if }}
                            {{ /collection:products #}}
                             {{#
                            <tr class="border-b border-gray-200">
                                <td class="text-sm px-6 py-4">
                                    <div class="flex flex-row items-center">
                                        <a class="hidden md:block" href="{{ product:url }}">
                                            {{ product:photos }}
                                                {{ if first }}
                                                    <img class="w-32"
                                                        src="{{ url }}" alt="{{ alt }}">
                                                {{ /if }}
                                            {{ /product:photos }}
                                        </a>
                                        <div class="md:px-6">
                                            <a class="font-bold" href="{{ product:url }}">
                                                {{ product:title }}
                                                {{ if variant }}({{ variant:variant }}){{ /if }}
                                            </a>

                                            <br>

                                            {{ sc:cart:removeItem :item="id" }}
                                                <button class="mt-1 block text-sm text-gray-700">Remove</button>
                                            {{ /sc:cart:removeItem }}
                                        </div>
                                    </div>
                                </td>

                                <td class="text-sm px-6 py-4">{{ variant:price ?? product:price }}</td>

                                <td class="text-sm px-6 py-4">
                                    {{ sc:cart:updateItem :item="id" }}
                                        <select
                                            class="bg-white p-1 rounded w-16"
                                            name="quantity"
                                            onchange="this.parentElement.submit()"
                                        >
                                            {{ loop from="1" to="5" }}
                                                <option
                                                    value="{{ value }}"
                                                    {{ if quantity == value }} selected {{ /if }}
                                                >
                                                    {{ value }}
                                                </option>
                                            {{ /loop }}
                                        </select>
                                    {{ /sc:cart:updateItem }}
                                </td>

                                <td class="text-sm px-6 py-4 text-right">{{ total }}</td>
                            </tr> #}}
                            
                        {{ /items }}

                        <tr>
                            <td></td>
                            <td></td>
                            <td class="text-sm px-6 py-3 font-bold">Subtotal</td>
                            <td class="text-sm px-6 py-3 text-right">{{ items_total }}</td>
                        </tr>
                    </tbody>
                </table>

                <div class="text-right text-sm px-6 text-gray-500">
                    <strong>Note:</strong> Les frais de livraisons et les taxes seront calculé au moment de passer à la caisse.
                </div>

                <div class="flex items-center justify-end my-12">
                    {{# partial:components/button label="Passer à la caisse" url="/checkout" #}}
                    <a href="/checkout" target="_self" class="inline-block font-bold focus:outline-none focus-visible:ring focus:ring-primary-900 focus:ring-opacity-20 focus-visible:ring-offset-2 transition-all duration-300 ease-in-out px-6 py-4 text-xl bg-primary-900 text-white hover:text-primary-900 hover:bg-white hover:border-2 hover:border-primary-900 border-2 border-transparent rounded-lg ">
                        Passer à la caisse
                    </a>
                </div>
            {{ /sc:cart }}
        {{ /if }}
    </div>
</main>

```

**Reply by mathamel** [reply]
I started from this but {{items}} only return the id

`<section class="fluid-container pt-12 text-center">
    {{ partial:typography/prose as="article" class="contents" }}
        <h1>Your Cart</h1>
    {{ /partial:typography/prose }}
</section>

<main>
    <div class="max-w-4xl mx-auto my-12 px-6">
        {{ partial:components/forms/errors_panel }}

        {{ if {sc:cart:has} === false || {sc:cart:count} === 0 }}
            <p class="text-center">Your cart is empty! <a href="{{ collection:pages id:is="6be90b19-994a-40f0-946d-0ce629d46940" }}{{ url }}{{/collection:pages }}">Start shopping</a>.</p>
        {{ else }}
            {{ sc:cart }}
                {{# if formation or private-training - go to checkout directly  #}}
                {{# items }}
                        {{ if {product.product_categories.slug} == "formations" or  {product.product_categories.slug} == "private-traiing"}}
                            {{ redirect to="{{site:url}}/checkout"}}
                        {{ /if }}
                {{ /items #}}

                {{# Nice Cards #}}
                <div class="md:hidden">
                    {{ items }}
                        

                        
                        <div class="flex flex-col items-center bg-white border border-gray-200 rounded-lg shadow flex-row max-w-xl hover:bg-gray-100 dark:border-gray-700 dark:bg-gray-800 dark:hover:bg-gray-700">
                            
                            <a class="" href="{{ product:url }}">
                                {{ product:photos }}
                                    {{ if first }}
                                        <img class="object-cover rounded-t-lg h-auto w-36 md:h-auto md:w-48 md:rounded-none md:rounded-s-lg"
                                            src="{{ url }}" alt="{{ alt }}">
                                    {{ /if }}
                                {{ /product:photos }}
                            </a>
                            <div class="flex flex-col justify-between p-4 leading-normal">
                                <h5 class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white">{{ product:title }}
                                    {{ if variant }}({{ variant:variant }}){{ /if }}</h5>


                                {{ if {product.product_categories.slug} == "formations" or  {product.product_categories.slug} == "private-traiing"}}
                                    {{ if product.start_date }}
                                    <p class="text-base mb-2"><i class="fa fa-calendar"></i> {{ product.start_date | format('Y/m/d') }} {{ if product.end_date }} to  {{ product.end_date | format('Y/m/d') }}{{ /if }}</p>
                                    {{ /if }}
                                {{ /if }}     
                                {{ variant:price ?? product:price }}     
                                {{ sc:cart:removeItem :item="id" }}
                                    <button class="mt-1 block text-sm text-gray-700">Remove</button>
                                {{ /sc:cart:removeItem }}
                            </div>
                        </div>


                   
                    {{ /items }}
                </div> 


            
                <table class="bg-white table-auto w-full rounded-lg p-4 mb-4 hidden md:block">
                    <thead>
                        <tr class="border-b border-gray-200">
                            <th class="px-6 py-3 text-left font-semibold text-sm w-2/3">Product</th>
                            <th class="px-6 py-3 text-left font-semibold text-sm">Price</th>
                            <th class="px-6 py-3 text-left font-semibold text-sm">Quantity</th>
                            <th class="px-6 py-3 text-right font-semibold text-sm">Total</th>
                        </tr>
                    </thead>

                    <tbody>
                        {{ items }}
                            <tr class="border-b border-gray-200">
                                <td class="text-sm px-6 py-4">
                                    <div class="flex flex-row items-center">
                                        <a class="hidden md:block" href="{{ product:url }}">
                                            {{ product:photos }}
                                                {{ if first }}
                                                    <img class="w-32"
                                                        src="{{ url }}" alt="{{ alt }}">
                                                {{ /if }}
                                            {{ /product:photos }}
                                        </a>
                                        <div class="md:px-6">
                                            <a class="font-bold" href="{{ product:url }}">
                                                {{ product:title }}
                                                {{ if variant }}({{ variant:variant }}){{ /if }}
                                            </a>

                                            <br>

                                            {{ sc:cart:removeItem :item="id" }}
                                                <button class="mt-1 block text-sm text-gray-700">Remove</button>
                                            {{ /sc:cart:removeItem }}
                                        </div>
                                    </div>
                                </td>

                                <td class="text-sm px-6 py-4">{{ variant:price ?? product:price }}</td>

                                <td class="text-sm px-6 py-4">
                                    {{ if {product.product_categories.slug} == "formations" or  {product.product_categories.slug} == "private-traiing"}}
                                            {{ quantity }}
                                    {{ else }}        
                                        {{ sc:cart:updateItem :item="id" }}
                                            <select
                                                class="bg-white p-1 rounded w-16"
                                                name="quantity"
                                                onchange="this.parentElement.submit()"
                                            >
                                                {{ loop from="1" to="5" }}
                                                    <option
                                                        value="{{ value }}"
                                                        {{ if quantity == value }} selected {{ /if }}
                                                    >
                                                        {{ value }}
                                                    </option>
                                                {{ /loop }}
                                            </select>
                                        {{ /sc:cart:updateItem }}
                                    {{ /if }}
                                    
                                </td>

                                <td class="text-sm px-6 py-4 text-right">{{ total }}</td>
                            </tr>
                        {{ /items }}

                        <tr>
                            <td></td>
                            <td></td>
                            <td class="text-sm px-6 py-3 font-bold">Subtotal</td>
                            <td class="text-sm px-6 py-3 text-right">{{ items_total }}</td>
                        </tr>
                    </tbody>
                </table>

                <div class="text-right text-sm px-6 text-gray-500 mt-8">
                    <strong>Note:</strong> Shipping & Tax will be calculated at the end of the checkout.
                </div>

                <div class="flex items-center justify-end my-12">
                    {{# partial:components/button label="Continue to Checkout" url="{{site:url}}/checkout" #}}
                    {{ partial:reflection/components/button label="Continue to Checkout" link="{{site:url}}/checkout" size="md"}}
                </div>
            {{ /sc:cart }}
        {{ /if }}
    </div>
</main>
`

**Reply by duncanmcclean** [reply]
Instead of `{{ items }}`, what happens if you try to use [`{{ sc:cart:items }}`](http://simple-commerce.duncanmcclean.com/tags/cart#content-line-items)?

**Reply by mathamel** [reply]
![image](https://github.com/user-attachments/assets/4d1c5a5d-4755-4eb8-bd54-32a1920f3ea6)
 the first line item, is from {{sc:cart:items}}
 
 the second line is using {{ sc:cart }} {{ items }} {{ collection:products id:is="{product}"}}
 
 but the Quantity dropdown is not working when using the 2nd way.

**Reply by duncanmcclean** [reply]
Are you able to replicate the issue on a fresh site using the Starter Kit?

**Reply by mathamel** [reply]
Hi, sorry it took longer before I had time to do this setup. I now face the same issues, here are the steps:

1. Create new Statamic app with the starter-kit 'statamic new mysite duncanmcclean/sc-starter-kit'
2. Install the Statamic Eloquent Driver (https://github.com/statamic/eloquent-driver) and import all types (Assets, Collections, Entries, etc.) into the database.
3. Switch Simple-Commerce to Database Orders (https://simple-commerce.duncanmcclean.com/database-orders#database-orders)

Now I get a cart without any data.

At step 1, everything works fine.
At step 2, cart do not function, orders are not format to fit into the database entry table, they do not share the same schema.
At step 3, cart/orders are saved, but products are not augmented.

![image](https://github.com/user-attachments/assets/88683ace-90dc-45f3-b785-c26b6147e667)

**Reply by mathamel** [reply]
Should I create an issue about this?

**Reply by duncanmcclean** [reply]
Can you share the site you were testing on? It'll make it easier for me to troubleshoot with.

**Reply by mathamel** [reply]
Can you send me your email, I'll be able to add you to the bitbucket repo and give you a database dump.

**Reply by duncanmcclean** [reply]
Send them to `duncan@doublethree.digital` and I'll take a look when I have some time.

**Reply by mathamel** [reply]
branch feature/ecommerce

**Reply by duncanmcclean** [reply]
Thanks - I'll take a look whenever I'm able.

**Reply by duncanmcclean** [reply]
I'm getting this error whenever I try to accept the invite:

![CleanShot 2024-09-18 at 18 02 26](https://github.com/user-attachments/assets/e917f1cf-77f9-4fe4-b1cb-1a7a929b8202)

**Comment by mathamel** [comment]
ok let me fix this



*Mathieu HamelPrésident514-360-0920 poste 101 | manaweb.ca
<http://manaweb.ca>*
<https://www.manaweb.ca/>
*Créativité • Agilité • Transparence *


Le mer. 18 sept. 2024 à 13:03, Duncan McClean ***@***.***> a
écrit :

**Comment by mathamel** [comment]
You can try again



*Mathieu HamelPrésident514-360-0920 poste 101 | manaweb.ca
<http://manaweb.ca>*
<https://www.manaweb.ca/>
*Créativité • Agilité • Transparence *


Le mer. 18 sept. 2024 à 13:03, Duncan McClean ***@***.***> a
écrit :

**Reply by duncanmcclean** [reply]
Thanks, it worked this time.

BTW: You posted the database link publicly on the GitHub discussion. I've deleted the comment - are you able to send it to me via email instead?

**Comment by mathamel** [comment]
Oh thanks for taking it down, didn't realise it was published publicly
there when replying via email.



*Mathieu HamelPrésident514-360-0920 poste 101 | manaweb.ca
<http://manaweb.ca>*
<https://www.manaweb.ca/>
*Créativité • Agilité • Transparence *


Le mer. 18 sept. 2024 à 14:09, Duncan McClean ***@***.***> a
écrit :

**Comment by duncanmcclean** [comment, marked as answer]
When you switch customers/orders to the database, it copies over the Runway blueprints. 

However, it was literally copying the YAML files, which obviously wouldn't do anything if your blueprints are stored in the database, which meant your customer/order blueprints were empty, so the product wasn't getting augmented.

I'm going to tag a fix shortly for new sites, however, to fix it on existing sites, you'll need to add this snippet to your `routes/console.php` file and run `php artisan fix-sc-blueprints`, then it should work as expected.

```php
Artisan::command('fix-sc-blueprints', function () {
    $stubsPath = __DIR__.'/../vendor/duncanmcclean/simple-commerce/src/Console/Commands/stubs';

    \Statamic\Facades\Blueprint::make('customers')
        ->setNamespace('runway')
        ->setContents(\Statamic\Facades\YAML::file($stubsPath.'/runway_customer_blueprint.yaml')->parse())
        ->save();

    \Statamic\Facades\Blueprint::make('orders')
        ->setNamespace('runway')
        ->setContents(\Statamic\Facades\YAML::file($stubsPath.'/runway_order_blueprint.yaml')->parse())
        ->save();
});
```
